src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerInfoCache.cs

```diff
@@ -114,5 +121,29 @@ public bool IsAnalyzerSuppressed(DiagnosticAnalyzer analyzer, Project project)
             //       and let the core analyzer driver in the compiler layer handle skipping redundant analysis callbacks.
             return false;
         }
+
+        public ISkippedAnalyzersInfo GetOrCreateSkippedAnalyzersInfo(Project project, HostDiagnosticAnalyzers hostAnalyzers)
+        {
+            if (_skippedAnalyzersInfo.TryGetValue(project.AnalyzerReferences, out var skippedAnalyzersInfo))
+            {
+                return skippedAnalyzersInfo;
+            }
+
+            var createValueCallback = new ConditionalWeakTable<IReadOnlyList<AnalyzerReference>, ISkippedAnalyzersInfo>.CreateValueCallback(
+                _ => SkippedHostAnalyzersInfo.Create(project, hostAnalyzers, this));
+            return _skippedAnalyzersInfo.GetValue(project.AnalyzerReferences, createValueCallback);
+        }
+
+        public ISkippedAnalyzersInfo GetOrCreateSkippedAnalyzersInfo(Project project, IEnumerable<AnalyzerReference> hostAnalyzers)
```

Comment:
This API is used for computing this info in OOP, as we don't have `HostDiagnosticAnalyzers` available in OOP.